lib/browser/api/menu.js

```diff
@@ -167,7 +167,9 @@ Menu.buildFromTemplate = function (template) {
     if (typeof item !== 'object') {
       throw new TypeError('Invalid template for MenuItem')
     }
-    menu.append(new MenuItem(item))
+    if( Object.keys(item).length > 0) {
```

Comment:
This won't actually check the JS object is a valid menu item, I could provide an object `{ thing: 'test' }` and this check would pass.

Not sure if that's a case we super want to handle, and it's not easy to handle as `MenuItem`'s options are all optional.

Also there's a weird space character here

Reply:
Thanks for the feedback, Would you prefer we check for only empty objects or would another solution be preferable?

Reply:
_All_ of the options are marked as `(optional)` in the documents, so allowing an empty menuitem matches the documentation. If we're going to remove that (mis)feature, we should document what the minimum option is for a MenuItem.

Maybe we document that at least one option is required, then call `menu.append()` only iff there is at least one key and _all_ keys are documented supported options

Reply:
I think that sounds reasonable, that way we can make things more consistent across platforms. I'll implement and update the docs.

Reply:
My concern with that is it's added maintenance every time we add a menu item option (which doesn't happen often but I see us forgetting to update this list of allowed properties).

I think we should figure out explicitly which options we should require, probably some group of "label", "role" and friends that actually make things render in the menu item.